src/containers/shared/components/TransactionActionIcon/TransactionActionIcon.tsx

```diff
@@ -0,0 +1,36 @@
+import { ReactElement } from 'react'
+import { TransactionAction } from '../Transaction/types'
+import { getAction } from '../Transaction'
+import TransactionCancelIcon from './TransactionCancelIcon.svg'
+import TransactionCreateIcon from './TransactionCreateIcon.svg'
+import TransactionFinishIcon from './TransactionFinishIcon.svg'
+import TransactionModifyIcon from './TransactionModifyIcon.svg'
+import TransactionSendIcon from './TransactionSendIcon.svg'
+import TransactionUnknownIcon from './TransactionUnknownIcon.svg'
+
+export interface TransactionActionIconProps {
+  action?: TransactionAction
+  type?: string
```

Comment:
Why are these both optional? Shouldn't they be mandatory?

Reply:
Because you can provide the action or the `TransactionType` which will derive the action for you.  One is required.

Reply:
Wouldn't it be better to do something like `type TransactionActionIconProps = { action: TransactionAction } | { type: string }`

Reply:
I get `TS2339: Property 'action' does not exist on type 'TransactionActionIconProps'.` on line 14 and 15 if I change it to that.

Had to update to

```
type TransactionActionIconProps =
  | { action: TransactionAction; type: never }
  | { action: never; type: string }
```